src/marqo/s2_inference/clip_utils.py

```diff
@@ -40,26 +40,29 @@ def format_and_load_CLIP_images(images: List[Union[str, ndarray, ImageType]]) ->
     
     return results
 
-def _load_image_from_path(image: str) -> ImageType:
-    """loads an image into PIL from a string path that is
-    either local or a url
+def load_image_from_path(image_path: str) -> ImageType:
+    """Loads an image into PIL from a string path that is either local or a url
 
     Args:
-        image (str): _description_
+        image_path (str): Local or remote path to image.
 
     Raises:
-        ValueError: _description_
+        ValueError: If the local path is invalid, or getting the image from a 
+        remote server returns a non-success status code.
 
     Returns:
-        ImageType: _description_
+        ImageType: In-memory PIL image.
     """
     
-    if os.path.isfile(image):
-        img = Image.open(image)
-    elif validators.url(image):
-        img = Image.open(requests.get(image, stream=True).raw)
+    if os.path.isfile(image_path):
+        img = Image.open(image_path)
+    elif validators.url(image_path):
+        resp = requests.get(image_path, stream=True)
+        if not resp.ok:
+            raise ValueError(f"image url {image_path} returned a {resp.status_code}. Reason {resp.reason}")
```

Comment:
it might be worth changing to a better excpetion.
PIL already has one we could use
`from PIL import UnidentifiedImageError`